lib/start-checking.js

```diff
@@ -6,12 +6,16 @@ internals.getOk = require('./get-ok')
 
 function startChecking (state, options) {
   options = parse(options)
-  handleInterval(state, options)
+  state !== undefined && state.method !== undefined && state.url !== undefined && handleInterval(state, options)
 }
 
 function handleInterval (state, options) {
   var timeout
   var ok = internals.getOk(state)
+  var timeoutHandler = function () {
```

Comment:
define as name function:

```js
function timeoutHandler () {
```

You can also define it outside

```js
function timeoutHandler (state, options) {
  var checkAgain = handleInterval.bind(null, state, options)
  internals.check(state, options).then(checkAgain, checkAgain)
}
```

and pass in state & options into [setTimeout](https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setTimeout) like

```js
state.checkTimeout = setTimeout(timeoutHandler, timeout, state, options)
```

Both ways are good with me, what ever you think makes the code more readable :)